File: ShoesStore/ShoesStore/Model/Product.cs
namespace ShoesStore.Model
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public List<ProductSize> ProductSizes { get; set; }
        public List<ProductImage> ProductImages { get; set; }
    }

}

File: UI/src/app/carts/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  categoryId: number;
  category: Category;
  createdAt: Date;
  updatedAt: Date;
  productSizeStocks: ProductSizeStock[];
  productImages: ProductImage[];
}

export interface Category {
  id: number;
  name: string;
}

export interface ProductSizeStock {
  sizeId: number;
  size: string;
  stock: number;
}

export interface ProductImage {
  id: number;
  url: string;
  productId: number;
}
